sdks/java/io/kafka/src/main/java/org/apache/beam/sdk/io/kafka/KafkaIO.java

```diff
@@ -1606,6 +1658,24 @@ public PCollection<KafkaRecord<K, V>> expand(PBegin input) {
                   .withMaxNumRecords(kafkaRead.getMaxNumRecords());
         }
 
+        if (kafkaRead.isRedistributed()) {
+          if (kafkaRead.isCommitOffsetsInFinalizeEnabled()) {
+            LOG.warn(
+                "Both isRedistribute and isCommitOffsetsInFinalizeEnabled are enabled, isRedistribute will take precendence");
```

Comment:
Redistribute is not guaranteed to do the checkpoint that is required by commitOffsetsInFinalize. Either we need to make both work correctly, or this should be a rejected configuration (throw an exception)

Reply:
Since its meant to be a hint to the runner, and the runner can choose what to do, I decided to fail this combination for now (instead of only failing if kafkaRead.isRedistributed()  + allowDuplicates ==true)